### PR TITLE
Update pipelines for Concourse 7.8.x

### PR DESCRIPTION
**Pull request by ashuka24**
This PR updates the PXF pipelines to remove deprecation warnings and ensure that pipelines can run without issues in a Concourse 7.8.x deployment. 

The changes made include: 
- Updating all group, job, task and resource names to be all lowercase
- Updating all spaces in said names to be underscores
- Removing parentheses
- Updating colon to dash in pipeline names

The pipeline states: 
| pipeline               | moved? | updated?   |
|------------------------|--------|------------|
| dev                    | yes    | yes        |
| pxf-build              | yes    | yes        |
| pr                     | yes    | yes        |
| certification          | yes    | yes        |
| cloudbuild             | yes    | yes        |
| pivnet                 | yes    | yes        |
| singlecluster          | yes    | yes        |
| hadoop-cluster-cleaner | yes    | not needed |
| perf pipelines         | no     | no         |
| pxf 5 pipelines        | no     | no         |

### CODE REVIEW

File: concourse/scripts/check_docker_images_and_tag.bash [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 set -euo pipefail
 
+IMAGE_TAG="${IMAGE_TAG:-latest}"
```
Comment (bradfordb-vmware) [inline concourse/scripts/check_docker_images_and_tag.bash:5]: Do we want to assume a default or error out if a value isn't given?

```sh
: "${IMAGE_TAG:?IMAGE_TAG must be set}"
```

File: concourse/scripts/check_docker_images_and_tag.bash [hunk 2]
```diff
@@ -39,13 +40,15 @@ do
 
     for image in "${IMAGE_NAMES[@]}"
     do
-      # we need to untag latest first
-      gcloud container images untag --quiet "gcr.io/${GOOGLE_PROJECT_ID}/gpdb-pxf-dev/${image}:latest" || true
-      # tag image with latest
-      echo "Tagging gcr.io/${GOOGLE_PROJECT_ID}/gpdb-pxf-dev/${image}:${COMMIT_SHA} with latest"
+      # we need to untag ${IMAGE_TAG} first
+      if [[ $IMAGE_TAG == "latest"]]; then
```
Comment (bradfordb-vmware) [inline concourse/scripts/check_docker_images_and_tag.bash:44]: I don't think this logic is correct -- there is nothing special about `latest` that requires us to untag. If there is an image with that tag already, then we need to untag. You can check if the image already contains a tag with

```sh
gcloud container images list-tags gcr.io/${GOOGLE_PROJECT_ID}/gpdb-pxf-dev/${image} --filter="tags : ${IMAGE_TAG}"
```

File: concourse/scripts/pxf_common.bash [hunk 3]
```diff
@@ -184,6 +184,9 @@ function build_install_gpdb() {
 }
 
 function install_gpdb_binary() {
+	# Ensure gpadmin owns its home directory
+	chown -R gpadmin:gpadmin /home/gpadmin
```
Comment (bradfordb-vmware) [inline concourse/scripts/pxf_common.bash:188]: Does this mean that we need to fix how our container image is built? It seems odd to me that ownership of a directory inside the container is being changed by Concourse.

**Review by bradfordb-vmware — COMMENTED** [review]
For the build_pipeline and dev_build_pipeline, I only reviewed from a very high-level and not in-depth.

File: concourse/scripts/pxf_common.bash (hunk 3 above)
Comment (yimingli-vmware) [reply concourse/scripts/pxf_common.bash:188]: Do we need to add a TODO to figure out what's happening behind?

**Review by yimingli-vmware — APPROVED** [review]

**Review by hpandeycodeit — APPROVED** [review]

**Review by yimingli-vmware — APPROVED** [review]

**Review by bradfordb-vmware — APPROVED** [review]

**Review by edespino — CHANGES_REQUESTED** [review]
Running yamllint (1.32.0) results in quite a few issues. It is possible these existed prior to this work. I prefer if these are cleaned up as part of this effort. What are your thoughts on this?

```
 ~/workspace/pxf/concourse  pr-1036 *6 !1 ?8  yamllint --version
yamllint 1.32.0
 ~/workspace/pxf/concourse  pr-1036 *6 !1 ?8  yamllint pipelines/cloudbuild_pipeline.yml
  4:1       warning  missing document start "---"  (document-start)
  5:3       error    wrong indentation: expected 0 but found 2  (indentation)
  19:9      error    wrong indentation: expected 6 but found 8  (indentation)
  30:9      error    wrong indentation: expected 6 but found 8  (indentation)
  45:3      error    wrong indentation: expected 0 but found 2  (indentation)
  64:11     error    wrong indentation: expected 8 but found 10  (indentation)
  86:121    error    line too long (158 > 120 characters)  (line-length)
  91:11     error    wrong indentation: expected 8 but found 10  (indentation)
  104:11    error    wrong indentation: expected 8 but found 10  (indentation)

 ~/workspace/pxf/concourse  pr-1036 *6 !1 ?8  yamllint pipelines/certification_pipeline.yml 
  73:6      warning  missing starting space in comment  (comments)

 ~/workspace/pxf/concourse  pr-1036 *6 !1 ?10  yamllint dev-release-eespino-pr-1036.yml
  111:7     error    wrong indentation: expected 4 but found 6  (indentation)
  128:7     error    wrong indentation: expected 4 but found 6  (indentation)
  245:7     error    wrong indentation: expected 4 but found 6  (indentation)
  376:1     error    trailing spaces  (trailing-spaces)
  393:1     error    trailing spaces  (trailing-spaces)
  517:1     error    trailing spaces  (trailing-spaces)
  653:1     error    trailing spaces  (trailing-spaces)
  717:1     error    trailing spaces  (trailing-spaces)
  725:1     error    trailing spaces  (trailing-spaces)
  906:1     error    trailing spaces  (trailing-spaces)
  1069:1    error    trailing spaces  (trailing-spaces)
  1086:16   error    too many spaces inside brackets  (brackets)
  1086:43   error    too many spaces inside brackets  (brackets)
  1251:43   warning  too few spaces before comment  (comments)
  1507:41   warning  too few spaces before comment  (comments)
  1590:11   warning  too few spaces before comment  (comments)
  1608:39   warning  too few spaces before comment  (comments)
  1880:9    warning  too few spaces before comment  (comments)
  2440:1    error    trailing spaces  (trailing-spaces)

 ~/workspace/pxf/concourse  pr-1036 *6 !1 ?10  yamllint dev-eespino-pr-1036.yml
> yamllint dev-eespino-pr-1036.yml
dev-eespino-pr-1036.yml
  12:7      error    wrong indentation: expected 4 but found 6  (indentation)
  29:7      error    wrong indentation: expected 4 but found 6  (indentation)
  64:1      error    too many blank lines (3 > 2)  (empty-lines)
  109:1     error    too many blank lines (3 > 2)  (empty-lines)
  137:1     error    trailing spaces  (trailing-spaces)
  167:1     error    too many blank lines (4 > 2)  (empty-lines)
  431:7     error    wrong indentation: expected 4 but found 6  (indentation)
  475:7     error    wrong indentation: expected 4 but found 6  (indentation)
  503:7     error    wrong indentation: expected 4 but found 6  (indentation)
  532:7     error    wrong indentation: expected 4 but found 6  (indentation)
  559:2     warning  comment not indented like content  (comments-indentation)
  566:1     error    too many blank lines (5 > 2)  (empty-lines)
  576:1     error    too many blank lines (6 > 2)  (empty-lines)
  581:2     warning  comment not indented like content  (comments-indentation)

 ~/workspace/pxf/concourse  pr-1036 *6 !1 ?10  
```

**Comment by edespino** [comment]
Nice work, I was able to successfully set the following pipelines without receiving any Concourse v7.8.3 deprecation warnings:

```
make set-dev-build-pipeline      FLY_CMD=fly783 CONCOURSE=ud2 FLY_OPTION_NON-INTERACTIVE=-n SLACK=false DEV_BUILD_PIPELINE_NAME=dev-build-eespino-pr-1036

make set-dev-release-pipeline    FLY_CMD=fly783 CONCOURSE=ud2 FLY_OPTION_NON-INTERACTIVE=-n SLACK=false DEV_BUILD_PIPELINE_NAME=dev-release-eespino-pr-1036

make set-certification-pipeline  FLY_CMD=fly783 CONCOURSE=ud2 FLY_OPTION_NON-INTERACTIVE=-n SLACK=false CERTIFICATION_PIPELINE_NAME=dev-certification-eespino-pr-1036

make set-dev-cloudbuild-pipeline FLY_CMD=fly783 CONCOURSE=ud2 FLY_OPTION_NON-INTERACTIVE=-n SLACK=false DEV_CLOUDBUILD_PIPELINE_NAME=dev-cloudbuild-eespino-pr-1036
```

**Comment by ashuka24** [comment]
> Running yamllint (1.32.0) results in quite a few issues. It is possible these existed prior to this work. I prefer if these are cleaned up as part of this effort. What are your thoughts on this?
> 
> ```
>  ~/workspace/pxf/concourse  pr-1036 *6 !1 ?8  yamllint --version
> yamllint 1.32.0
>  ~/workspace/pxf/concourse  pr-1036 *6 !1 ?8  yamllint pipelines/cloudbuild_pipeline.yml
>   4:1       warning  missing document start "---"  (document-start)
>   5:3       error    wrong indentation: expected 0 but found 2  (indentation)
>   19:9      error    wrong indentation: expected 6 but found 8  (indentation)
>   30:9      error    wrong indentation: expected 6 but found 8  (indentation)
>   45:3      error    wrong indentation: expected 0 but found 2  (indentation)
>   64:11     error    wrong indentation: expected 8 but found 10  (indentation)
>   86:121    error    line too long (158 > 120 characters)  (line-length)
>   91:11     error    wrong indentation: expected 8 but found 10  (indentation)
>   104:11    error    wrong indentation: expected 8 but found 10  (indentation)
> 
>  ~/workspace/pxf/concourse  pr-1036 *6 !1 ?8  yamllint pipelines/certification_pipeline.yml 
>   73:6      warning  missing starting space in comment  (comments)
> 
>  ~/workspace/pxf/concourse  pr-1036 *6 !1 ?10  yamllint dev-release-eespino-pr-1036.yml
>   111:7     error    wrong indentation: expected 4 but found 6  (indentation)
>   128:7     error    wrong indentation: expected 4 but found 6  (indentation)
>   245:7     error    wrong indentation: expected 4 but found 6  (indentation)
>   376:1     error    trailing spaces  (trailing-spaces)
>   393:1     error    trailing spaces  (trailing-spaces)
>   517:1     error    trailing spaces  (trailing-spaces)
>   653:1     error    trailing spaces  (trailing-spaces)
>   717:1     error    trailing spaces  (trailing-spaces)
>   725:1     error    trailing spaces  (trailing-spaces)
>   906:1     error    trailing spaces  (trailing-spaces)
>   1069:1    error    trailing spaces  (trailing-spaces)
>   1086:16   error    too many spaces inside brackets  (brackets)
>   1086:43   error    too many spaces inside brackets  (brackets)
>   1251:43   warning  too few spaces before comment  (comments)
>   1507:41   warning  too few spaces before comment  (comments)
>   1590:11   warning  too few spaces before comment  (comments)
>   1608:39   warning  too few spaces before comment  (comments)
>   1880:9    warning  too few spaces before comment  (comments)
>   2440:1    error    trailing spaces  (trailing-spaces)
> 
>  ~/workspace/pxf/concourse  pr-1036 *6 !1 ?10  yamllint dev-eespino-pr-1036.yml
> > yamllint dev-eespino-pr-1036.yml
> dev-eespino-pr-1036.yml
>   12:7      error    wrong indentation: expected 4 but found 6  (indentation)
>   29:7      error    wrong indentation: expected 4 but found 6  (indentation)
>   64:1      error    too many blank lines (3 > 2)  (empty-lines)
>   109:1     error    too many blank lines (3 > 2)  (empty-lines)
>   137:1     error    trailing spaces  (trailing-spaces)
>   167:1     error    too many blank lines (4 > 2)  (empty-lines)
>   431:7     error    wrong indentation: expected 4 but found 6  (indentation)
>   475:7     error    wrong indentation: expected 4 but found 6  (indentation)
>   503:7     error    wrong indentation: expected 4 but found 6  (indentation)
>   532:7     error    wrong indentation: expected 4 but found 6  (indentation)
>   559:2     warning  comment not indented like content  (comments-indentation)
>   566:1     error    too many blank lines (5 > 2)  (empty-lines)
>   576:1     error    too many blank lines (6 > 2)  (empty-lines)
>   581:2     warning  comment not indented like content  (comments-indentation)
> 
>  ~/workspace/pxf/concourse  pr-1036 *6 !1 ?10  
> ```

We were able to remove most of the yamllint errors seen here. There are a couple left (line too long) that cannot be updated until we change the way we pass in list-type variables, which is out of scope of this PR

**Comment by edespino** [comment]
The following addresses to remaining errors:

```
modified   concourse/pipelines/cloudbuild_pipeline.yml
@@ -84,6 +84,7 @@ jobs:
       GOOGLE_CREDENTIALS: ((ud/pxf/secrets/pxf-cloudbuild-service-account-key))
       GOOGLE_PROJECT_ID: ((ud/pxf/common/google-project-id))
       GOOGLE_ZONE: ((ud/pxf/common/google-zone))
+      # yamllint disable rule:line-length
       IMAGE_LIST: "gpdb5-centos7-test-pxf gpdb6-centos7-test-pxf gpdb6-rocky8-test-pxf gpdb6-ubuntu18.04-test-pxf gpdb6-oel7-test-pxf gpdb7-rocky8-test-pxf"
       IMAGE_TAG: ((gcp-image-tag))
     config:
modified   concourse/pipelines/templates/dev_build_pipeline-tpl.yml
@@ -1,4 +1,6 @@
 ---
+# yamllint disable rule:empty-lines
+
 ## ======================================================================
 ## ANCHORS
 ## ======================================================================
```

**Review by edespino — APPROVED** [review]
Thank you for incorporating additional change requests.

**Pull request merged**